File: windows-companion/RecruitmentCore/RecruitmentData.cs
namespace RecruitmentCore;

// Placeholder dataset -- mirrors the Android project's RecruitmentData.kt 1:1
// (same tag IDs/names, same fake operators) so both apps agree until the real
// tag list and operator data are available. Swap this out once they are.
public static class RecruitmentData
{
    public static readonly IReadOnlyList<Tag> AllTags = new List<Tag>
    {
        new(1, "Melee", TagCategory.Position),
        new(2, "Ranged", TagCategory.Position),
        new(3, "Vanguard", TagCategory.Class),
        new(4, "Guard", TagCategory.Class),
        new(5, "Sniper", TagCategory.Class),
        new(6, "Defender", TagCategory.Class),
        new(7, "Medic", TagCategory.Class),
        new(8, "Supporter", TagCategory.Class),
        new(9, "Caster", TagCategory.Class),
        new(10, "Specialist", TagCategory.Class),
        new(11, "Starter", TagCategory.Qualification),
        new(12, "Senior Operator", TagCategory.Qualification),
        new(13, "Top Operator", TagCategory.Qualification),
        new(14, "Crowd-Control", TagCategory.Affix),
        new(15, "Nuker", TagCategory.Affix),
        new(16, "Healing", TagCategory.Affix),
        new(17, "Support", TagCategory.Affix),
        new(18, "DP-Recovery", TagCategory.Affix),
        new(19, "DPS", TagCategory.Affix),
        new(20, "Survival", TagCategory.Affix),
        new(21, "Debuff", TagCategory.Affix),
        new(22, "Fast-Redeploy", TagCategory.Affix),
        new(23, "AoE", TagCategory.Affix),
        new(24, "Defense", TagCategory.Affix),
        new(25, "Slow", TagCategory.Affix),
        new(26, "Shift", TagCategory.Affix),
        new(27, "Summon", TagCategory.Affix),
        new(28, "Robot", TagCategory.Affix),
        new(29, "Elemental", TagCategory.Affix),
    };

    public static readonly IReadOnlyList<Operator> AllOperators = new List<Operator>
    {
        new("Sample Op A", 1, new HashSet<int> { 1, 3 }),
        new("Sample Op B", 2, new HashSet<int> { 1 }),
        new("Sample Op C", 3, new HashSet<int> { 2 }),
        new("Sample Op D", 4, new HashSet<int> { 2, 3 }),
        new("Sample Op E", 4, new HashSet<int> { 21 }),
        new("Sample Op F", 5, new HashSet<int> { 21 }),
        new("Sample Op G", 4, new HashSet<int> { 22 }),
        new("Sample Op H", 5, new HashSet<int> { 22 }),
        new("Sample Op I", 5, new HashSet<int> { 21, 22 }),
    };
}

File: app/src/main/java/com/arknights/recruitmenthelper/data/RecruitmentData.kt
package com.arknights.recruitmenthelper.data

// Placeholder dataset. Swap ALL_TAGS/ALL_OPERATORS for the real tag list and a
// bundled operator list (e.g. loaded from an assets/operators.json) once available —
// RecruitmentCalculator only depends on the Tag/Operator shapes, not on how they're sourced.
object RecruitmentData {

    val ALL_TAGS = listOf(
        Tag(1, "Melee", TagCategory.POSITION),
        Tag(2, "Ranged", TagCategory.POSITION),
        Tag(3, "Vanguard", TagCategory.CLASS),
        Tag(4, "Guard", TagCategory.CLASS),
        Tag(5, "Sniper", TagCategory.CLASS),
        Tag(6, "Defender", TagCategory.CLASS),
        Tag(7, "Medic", TagCategory.CLASS),
        Tag(8, "Supporter", TagCategory.CLASS),
        Tag(9, "Caster", TagCategory.CLASS),
        Tag(10, "Specialist", TagCategory.CLASS),
        Tag(11, "Starter", TagCategory.QUALIFICATION),
        Tag(12, "Senior Operator", TagCategory.QUALIFICATION),
        Tag(13, "Top Operator", TagCategory.QUALIFICATION),
        Tag(14, "Crowd-Control", TagCategory.AFFIX),
        Tag(15, "Nuker", TagCategory.AFFIX),
        Tag(16, "Healing", TagCategory.AFFIX),
        Tag(17, "Support", TagCategory.AFFIX),
        Tag(18, "DP-Recovery", TagCategory.AFFIX),
        Tag(19, "DPS", TagCategory.AFFIX),
        Tag(20, "Survival", TagCategory.AFFIX),
        Tag(21, "Debuff", TagCategory.AFFIX),
        Tag(22, "Fast-Redeploy", TagCategory.AFFIX),
        Tag(23, "AoE", TagCategory.AFFIX),
        Tag(24, "Defense", TagCategory.AFFIX),
        Tag(25, "Slow", TagCategory.AFFIX),
        Tag(26, "Shift", TagCategory.AFFIX),
        Tag(27, "Summon", TagCategory.AFFIX),
        Tag(28, "Robot", TagCategory.AFFIX),
        Tag(29, "Elemental", TagCategory.AFFIX),
    )

    // Just enough fake operators to exercise every branch of the calculator:
    // - Melee/Vanguard alone pull in a 1-star op, so neither is "guaranteed".
    // - Debuff alone and Fast-Redeploy alone are each guaranteed 4-star.
    // - Ranged+Vanguard only intersects on one 4-star op -> guaranteed 4-star combo.
    // - Debuff+Fast-Redeploy only intersects on one 5-star op -> guaranteed 5-star combo,
    //   which should outrank all of the above when evaluated together.
    val ALL_OPERATORS = listOf(
        Operator("Sample Op A", rarity = 1, tagIds = setOf(1, 3)),
        Operator("Sample Op B", rarity = 2, tagIds = setOf(1)),
        Operator("Sample Op C", rarity = 3, tagIds = setOf(2)),
        Operator("Sample Op D", rarity = 4, tagIds = setOf(2, 3)),
        Operator("Sample Op E", rarity = 4, tagIds = setOf(21)),
        Operator("Sample Op F", rarity = 5, tagIds = setOf(21)),
        Operator("Sample Op G", rarity = 4, tagIds = setOf(22)),
        Operator("Sample Op H", rarity = 5, tagIds = setOf(22)),
        Operator("Sample Op I", rarity = 5, tagIds = setOf(21, 22)),
    )
}
